src/components/LayoutTree.tsx

```diff
@@ -599,8 +644,36 @@ const LayoutTree = () => {
       }
     })
 
-    return { indentCursorAncestorTables, treeThoughtsPositioned }
-  }, [fontSize, sizes, singleLineHeight, treeThoughts])
+    // If insertionY is still null, newRank would be inserted at the end
+    if (isSortedContext && insertionY === null && treeThoughtsPositioned.length > 0 && sourceThought) {
+      const lastThought = treeThoughtsPositioned[treeThoughtsPositioned.length - 1]
+      insertionY = lastThought.y + lastThought.height
+    }
+
+    // Determine hoverArrowVisibility based on insertionY
+    if (insertionY !== null) {
+      if (insertionY > maxVisibleY) {
+        hoverArrowVisibility = 'below'
+      } else if (insertionY < scrollTop - (toolbarRef.current ?? 0)) {
```

Comment:
@trevinhofmann  I do have a solution where we match the ranks of first and last thought with the `newRank` instead of checking the y values, while it works the only problem I see is there are edge cases when there is a little space left on top of the first visible thought, just enough for drophover to show, but since the rank will be generated for above the first thought arrow will render, like in the video below. We still would need to use the toolbar height in this case though.

[Screencast from 10-17-2024 01:43:59 PM.webm](https://github.com/user-attachments/assets/b1fb9c5e-8baf-4ca1-924c-ce33dc85a834)

Reply:
I'm okay with that! While technically the hover is in the viewport, it's at the very top of the screen, above all visible thoughts, so the arrow does not feel inappropriate. Thanks for the attention to detail.

Reply:
Understood, there are a couple of more edge cases that also needs to be addressed, like when last visible thought is also the last thought in context, same for first, the drop position rank is after the last/first, even in those cases arrow would render. I believe these needs to be fixed with this approach.

[Screencast from 10-17-2024 04:19:26 PM.webm](https://github.com/user-attachments/assets/10b6c338-9092-47b1-8c72-f578e71cba35)

Reply:
Yes, that one should be addressed. It doesn't look right if the arrow appears when the DropHover is well within the viewport.

Reply:
The arrow does still appear when the thought above/below the target (Louisiana in this example) is _partially_ in view. Personally, I think that is still acceptable, but I wanted to make sure we're aware of it:

https://github.com/user-attachments/assets/118db3bb-02a8-4bde-beff-5e38f720456f